LeagueToolkit/IO/TEXFile/TEX.cs

```diff
@@ -58,32 +72,40 @@ public TEX(Stream stream)
 
             stream.Seek(1, SeekOrigin.Current); // unknown, always 0
             texHeader.hasMipmaps = br.ReadBoolean();
-            Header = texHeader;
+            this.Header = texHeader;
 
-            MipMapCount = Header.hasMipmaps ? (int)Math.Log(Math.Max(Header.width, Header.height), 2) : 0;
-            if (MipMapCount > 0)
+            this.MipMapCount = Header.hasMipmaps ? (int)Math.Log(Math.Max(Header.width, Header.height), 2) : 0;
+            if (this.MipMapCount > 0)
             {
-                MipMapsBuffer = new byte[MipMapCount][];
+                MipMapsBuffer = new byte[this.MipMapCount][];
                 // mipmaps are written in order from smallest to largest
-                for (int i = MipMapCount; i > 0; i--)
+                for (int i = this.MipMapCount; i > 0; i--)
                 {
-                    int currentWidth = Math.Max(Header.width / (1 << i), MIN_WIDTH);
-                    int currentHeight = Math.Max(Header.height / (1 << i), MIN_HEIGHT);
-                    MipMapsBuffer[i - 1] = br.ReadBytes(currentWidth * currentHeight);
+                    int currentWidth = Math.Max(this.Header.width / (1 << i), 1);
+                    int currentHeight = Math.Max(this.Header.height / (1 << i), 1);
+                    int currentSize = Math.Max(currentWidth * currentHeight * BytesPerBlock(this.Header.format) /
```

Comment:
I suggest calling the `BytesPerBlock` and `BlockSize` functions once and storing the value